UnityProject/Assets/Scripts/Systems/Inventory/Pickupable.cs

```diff
@@ -146,29 +147,41 @@ public void ServerRollbackClient(HandApply interaction)
 	}
 
 	public virtual void ServerPerformInteraction(HandApply interaction)
+	{
+		StartCoroutine(serverPreformInteractionLogic(interaction));
+	}
+
+	private IEnumerator serverPreformInteractionLogic(HandApply interaction)
 	{
 		//we validated, but object may only be in extended range
 		var cnt = GetComponent<CustomNetTransform>();
 		var ps = interaction.Performer.GetComponent<PlayerScript>();
 		var extendedRangeOnly = !ps.IsRegisterTileReachable(cnt.RegisterTile, true);
 
+		float animTime = 0.2f;
```

Comment:
make it a constant of the class or you could even make it a serialized field if we wanted to change the speed eventually.